Dockerfile

```diff
@@ -81,12 +77,29 @@ RUN wget ${PYENV_REPO}/archive/v${PYENV_RELEASE}.zip \
   && mv $PYENV_ROOT/pyenv-$PYENV_RELEASE/* $PYENV_ROOT/ \
   && rm -r $PYENV_ROOT/pyenv-$PYENV_RELEASE
 
-ENV PATH $PYENV_ROOT/bin:$PATH
-
+#
+# Uncomment these lines if you just want to install python...
+#
+# ENV PATH $PYENV_ROOT/bin:$PYENV_ROOT/versions/${PYENV_PYTHON_VERSION}/bin:$PATH
+# RUN echo 'eval "$(pyenv init -)"' >> /etc/profile \
+#     && eval "$(pyenv init -)" \
+#     && pyenv install $PYENV_PYTHON_VERSION \
+#     && pyenv local ${PYENV_PYTHON_VERSION}
+
+#
+# ...uncomment these lines if you want to also debug python code in GDB
+#
+ENV PATH $PYENV_ROOT/bin:$PYENV_ROOT/versions/${PYENV_PYTHON_VERSION}-debug/bin:$PATH
 RUN echo 'eval "$(pyenv init -)"' >> /etc/profile \
     && eval "$(pyenv init -)" \
-    && pyenv install $PYENV_PYTHON_VERSION \
-    && pyenv local $PYENV_PYTHON_VERSION
+    && pyenv install --debug --keep $PYENV_PYTHON_VERSION \
+    && pyenv local ${PYENV_PYTHON_VERSION}-debug
+RUN ln -s /opt/pyenv/sources/${PYENV_PYTHON_VERSION}-debug/Python-${PYENV_PYTHON_VERSION}/python-gdb.py /opt/pyenv/versions/${PYENV_PYTHON_VERSION}-debug/bin/python3.6-gdb.py
+RUN ln -s /opt/pyenv/sources/${PYENV_PYTHON_VERSION}-debug/Python-${PYENV_PYTHON_VERSION}/python-gdb.py /opt/pyenv/versions/${PYENV_PYTHON_VERSION}-debug/bin/python3-gdb.py
+RUN ln -s /opt/pyenv/sources/${PYENV_PYTHON_VERSION}-debug/Python-${PYENV_PYTHON_VERSION}/python-gdb.py /opt/pyenv/versions/${PYENV_PYTHON_VERSION}-debug/bin/python-gdb.py
+RUN apt-get -qq update && \
+    apt-get -qq --yes --no-install-recommends --no-install-suggests install gdb
+RUN echo add-auto-load-safe-path /opt/pyenv/sources/${PYENV_PYTHON_VERSION}-debug/Python-${PYENV_PYTHON_VERSION}/ >> etc/gdb/gdbinit
```

Comment:
Thanks to @jsf9k for his work in figuring out how to create a `gdb`-ready version of Python that is installed via pyenv, in the Docker container.